judge/models/tests/util.py

```diff
@@ -0,0 +1,330 @@
+                user_permissions=('see_private_contest',),
+            ),
+            'staff_contest_edit_all': create_user(
+                username='staff_contest_edit_all',
+                is_staff=True,
+                user_permissions=('edit_own_contest', 'edit_all_contest'),
+            ),
+            'staff_problem_edit_own': create_user(
+                username='staff_problem_edit_own',
+                is_staff=True,
+                user_permissions=('edit_own_problem', 'rejudge_submission'),
+            ),
+            'staff_problem_see_all': create_user(
+                username='staff_problem_see_all',
+                user_permissions=('see_private_problem',),
+            ),
+            'staff_problem_edit_all': create_user(
+                username='staff_problem_edit_all',
+                is_staff=True,
+                user_permissions=('edit_own_problem', 'edit_all_problem'),
+            ),
+            'staff_problem_edit_public': create_user(
+                username='staff_problem_edit_public',
+                is_staff=True,
+                user_permissions=('edit_own_problem', 'edit_public_problem'),
+            ),
+            'staff_problem_see_organization': create_user(
+                username='staff_problem_see_organization',
+                user_permissions=('see_organization_problem',),
+            ),
+            'staff_problem_edit_all_with_rejudge': create_user(
+                username='staff_problem_edit_all_with_rejudge',
+                is_staff=True,
+                user_permissions=('edit_own_problem', 'edit_all_problem', 'rejudge_submission'),
+            ),
+            'staff_problem_edit_own_no_staff': create_user(
+                username='staff_problem_edit_own_no_staff',
+                user_permissions=('edit_own_problem', 'rejudge_submission'),
+            ),
+            'staff_submission_view_all': create_user(
+                username='staff_submission_view_all',
+                is_staff=True,
+                user_permissions=('view_all_submission',),
+            ),
+            'normal': create_user(
+                username='normal',
+            ),
+            'anonymous': AnonymousUser(),
+        }
+
+        self.profiles = {
+            # unauthenticated users have no profile
+            username: getattr(user, 'profile', None) for username, user in self.users.items()
+        }
+
+        self.organizations = {
+            'open': create_organization(
+                name='open',
+                registrant=self.profiles['superuser'],
+            ),
+        }
+
+    def _test_object_user(self, obj, user, methods):
+        for method, func in methods.items():
+            with self.subTest(method=method):
+                func(
+                    getattr(obj, method)(user),
+                    msg='Method "%s" failed for user "%s", object "%s".' % (method, user.username, obj),
+                )
+
+    def _test_object(self, obj, data):
```

Comment:
This name is too generic. This is really for testing methods that takes a user, no?

Reply:
Updated the name to `_test_object_methods_with_users`